pkg/backup/pvc_snapshot_tracker.go

```diff
@@ -20,45 +20,60 @@ import (
 	"fmt"
 
 	corev1api "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/util/sets"
 )
 
 // pvcSnapshotTracker keeps track of persistent volume claims that have been snapshotted
 // with pod volume backup.
 type pvcSnapshotTracker struct {
-	pvcs sets.String
+	pvcs map[string]pvcSnapshotStatus
+}
+
+type pvcSnapshotStatus struct {
+	taken bool
 }
 
 func newPVCSnapshotTracker() *pvcSnapshotTracker {
 	return &pvcSnapshotTracker{
-		pvcs: sets.NewString(),
+		pvcs: make(map[string]pvcSnapshotStatus),
 	}
 }
 
-// Track takes a pod and a list of volumes from that pod that were snapshotted, and
-// tracks each snapshotted volume that's a PVC.
-func (t *pvcSnapshotTracker) Track(pod *corev1api.Pod, snapshottedVolumes []string) {
-	for _, volumeName := range snapshottedVolumes {
-		// if the volume is a PVC, track it
-		for _, volume := range pod.Spec.Volumes {
-			if volume.Name == volumeName {
-				if volume.PersistentVolumeClaim != nil {
-					t.pvcs.Insert(key(pod.Namespace, volume.PersistentVolumeClaim.ClaimName))
+// Track indicates a volume from a pod should be snapshotted by pod volume backup.
+func (t *pvcSnapshotTracker) Track(pod *corev1api.Pod, volumeName string) {
```

Comment:
Track .vs. Take is a little confusing, how about pass the status or mode as a parameter and make the renaming more exlicit?

so the caller will call `tracker.Track(pod, volume, VOLUME_SNAPSHOT)` or `tracker.Track(pod, volume, PODVOLUME_BACKUP)`

Reply:
I have been thinking of how to clearly distinguish them, actually:

- Track means the volume is decided to be handled by PVB according to rules and annotations. We need to record this situation because we don't want the BIA and volume snapshotter to process the volume again
- Take means the volume has been processed by PVB successfully. We need to record this situation, because we want the volumes that is to processed by PVB but missed by the backup with the 1st pod should be processed by other pods. (In the case of RWX mount)

Therefore, if we want to use status, the status should be something like:

- VestedToPVB
- ProcessedByPVB

What do you think? @reasonerjt